src/components/LayoutTree.tsx

```diff
@@ -567,12 +573,16 @@ const LayoutTree = () => {
         ycol1Ancestors.push({ y: yaccum + height, depth: node.depth })
       }
 
+      const nextAutofocus = next?.autofocus
+      const nextIsLastVisible = nextAutofocus === 'show' || nextAutofocus === 'dim'
+
       return {
         ...node,
         cliff,
         height,
         singleLineHeightWithCliff,
         width: tableCol1Widths.get(head(parentOf(node.path))),
+        isLastVisible: !nextIsLastVisible,
```

Comment:
Won't this incorrectly match any thought that precedes a visible thought? We only want to match the last visible thought.

Reply:
The idea was that if the next thought is `undefined` or autofocus 'hide' then the current is the last visible thought here.

Reply:
As written, this will evaluate to true for more than just the last visible thought. Here's one case (but there may be others):

```
- a
  - b
    - c
- x
```

Set the cursor on `c`.

`isLastVisible` evaluates to true for both `c` and `x`.

Reply:
Right, I understand now. We can add a check for `node.autofocus !== 'hide'` but if there are more thoughts below the last visible their isLastVisible would still be set to `true`.

Reply:
Only one thought can be visible and followed by a hidden thought, as **em** does not render non-contiguous visible thoughts. Once you have the desired logic set up, I would suggest testing it on a few different cases to ensure it works as expected.